parser.go

```diff
@@ -2,54 +2,83 @@ package main
 
 import (
 	"bufio"
-	"fmt"
 	"io"
-	"log"
 	"regexp"
+	"runtime"
 	"strconv"
 )
 
 const (
-	GCRegexp   = `gc\d+\(\d+\): \d+\+\d+\+\d+\+\d+ us, \d+ -> (?P<Heap1>\d+) MB, \d+ \(\d+-\d+\) objects,( \d+ goroutines,)? \d+\/\d+\/\d+ sweeps, \d+\(\d+\) handoff, \d+\(\d+\) steal, \d+\/\d+\/\d+ yields`
-	SCVGRegexp = `scvg\d+: inuse: (?P<inuse>\d+), idle: (?P<idle>\d+), sys: (?P<sys>\d+), released: (?P<released>\d+), consumed: (?P<consumed>\d+) \(MB\)`
+	GCRegexpGo14 = `gc\d+\(\d+\): ([\d.]+\+?)+ us, \d+ -> (?P<Heap1>\d+) MB, \d+ \(\d+-\d+\) objects,( \d+ goroutines,)? \d+\/\d+\/\d+ sweeps, \d+\(\d+\) handoff, \d+\(\d+\) steal, \d+\/\d+\/\d+ yields`
+	GCRegexpGo15 = `gc \d+ @[\d.]+s \d+%: ([\d.]+\+?)+ ms clock, ([\d.+\/]+?)+ ms cpu, \d+->\d+->\d+ MB, (?P<Heap1>\d+) MB goal, \d+ P`
+	SCVGRegexp   = `scvg\d+: inuse: (?P<inuse>\d+), idle: (?P<idle>\d+), sys: (?P<sys>\d+), released: (?P<released>\d+), consumed: (?P<consumed>\d+) \(MB\)`
 )
 
 var (
-	gcre   = regexp.MustCompile(GCRegexp)
-	scvgre = regexp.MustCompile(SCVGRegexp)
+	gcrego14 = regexp.MustCompile(GCRegexpGo14)
+	gcrego15 = regexp.MustCompile(GCRegexpGo15)
+	scvgre   = regexp.MustCompile(SCVGRegexp)
 )
 
 type Parser struct {
-	reader   io.Reader
-	gcChan   chan *gctrace
-	scvgChan chan *scvgtrace
+	reader      io.Reader
+	GcChan      chan *gctrace
+	ScvgChan    chan *scvgtrace
+	NoMatchChan chan string
+	done        chan bool
+
+	Err error
+
+	gcRegexp   *regexp.Regexp
+	scvgRegexp *regexp.Regexp
+}
+
+func NewParser(r io.Reader) *Parser {
+	return &Parser{
+		reader:      r,
+		GcChan:      make(chan *gctrace, 1),
+		ScvgChan:    make(chan *scvgtrace, 1),
+		NoMatchChan: make(chan string, 1),
+		done:        make(chan bool, 1),
+	}
 }
 
 func (p *Parser) Run() {
 	sc := bufio.NewScanner(p.reader)
 
+	if p.gcRegexp == nil {
+		// Set regexp based on Golang version
+		if runtime.Version() == "go1.5" {
```

Comment:
This will fail if the runtime version is not go1.5, for example
```
lucky(~/src) % ~/go1.5/bin/go run version.go
go1.5.1
```
You could use a conditional compilation, ie break it out into two files with build tags // +build !go1.4, etc, or use strings.HasPrefix